packages/amplify-util-mock/src/CFNParser/intrinsic-functions.ts

```diff
@@ -0,0 +1,277 @@
+import { CloudFormationParseContext } from './types';
+import { isPlainObject, isInteger } from 'lodash';
+import { isString, isArray, isObject } from 'util';
+
+export function cfnJoin(
+  valNode: [string, string[]],
+  { params, conditions, resources, exports }: CloudFormationParseContext,
+  processValue
+) {
+  if (!(Array.isArray(valNode) && valNode.length === 2)) {
+    throw new Error(
+      `FN::Join expects an array with 2 elements instead got ${JSON.stringify(valNode)}`
+    );
+  }
+  const delimiter = valNode[0];
+  const items = valNode[1].map(item =>
+    processValue(item, { params, conditions, resources, exports })
+  );
+  return items.join(delimiter);
+}
+
+export function cfnSub(
+  valNode,
```

Comment:
why not use tuple type ```[string, object]``` for ```valNode```? Similar in other functions, will not adding comment there as well.